src/Adapters/Storage/Postgres/PostgresStorageAdapter.js

```diff
@@ -2523,9 +2523,41 @@ export class PostgresStorageAdapter implements StorageAdapter {
     return result;
   }
 
-  // TODO: implement?
-  ensureIndex(): Promise<void> {
-    return Promise.resolve();
+  async ensureIndex(
+    className: string,
+    schema: SchemaType,
+    fieldNames: string[],
+    indexName: ?string,
+    caseInsensitive: boolean = false
+  ): Promise<any> {
+    const defaultIndexName = `parse_default_${fieldNames.sort().join('_')}`;
+    const indexNameOptions: Object = indexName ? { name: indexName } : { name: defaultIndexName };
+    const constraintPatterns =  caseInsensitive ? fieldNames.map(
+      (fieldName, index) => `lower($${index + 3}:name) varchar_pattern_ops`
+    ) : fieldNames.map(
+      (fieldName, index) => `$${index + 3}:name`
+    );
+    const qs = `CREATE INDEX $1:name ON $2:name (${constraintPatterns.join()})`;
+    return this._client.none(qs, [indexName, className, ...fieldNames])
```

Comment:
I think the issue with the test failing is how this returns. It adds the index, but I did something wrong with the promise